src/util/expr.h

```diff
@@ -318,6 +318,40 @@ class exprt:public irept
   const_unique_depth_iteratort unique_depth_cend() const;
 };
 
+/// Protected class for all expressions.
+/// This API will eventually replace exprt.
+class protected_exprt : public exprt
```

Comment:
:pick: I slightly worry about the name, in general, `X_exprt` means an expression representing an AST node of type X. I guess `protected` is never a node of a C/C++/Java AST, but perhaps this should be called `expr_protected` or `expr_api_protected` to indicate that this is an internal fudge rather than an actual GOTO thing.

Reply:
ok